src/react-vite/08-stateful-hooks/stateful-hooks.md

```diff
@@ -2,51 +2,117 @@
 @parent learn-react-vite 8
 @outline 3
 
-@description TODO
+@description Work with React's useState hook to manage a component's state.
 
 @body
 
 ## Overview
 
-TODO
+When users make choices that need to be maintained or that affect other parts of
+the UI, we need to use "state." This lesson introduces React hooks that manage
+component state.
+
+## Objective: Manage component state using hooks
+
+React provides hooks that simplify managing state in a component. We will use
+some of these hooks to store the user's selections for state and city.
+
+> Note: Hooks must always appear in the body of the component **before** any
+> statements that might change the component's behavior. Place hooks at the top
+> of a component's body.
+
+### useState
+
+We can store state that persists through component rendering with the `useState`
+hook. You can set the initial state value when the component **first** renders
+by providing the value as an argument to the hook. _If you do not provide a
+value the initial state value will be `undefined`._
+
+This example shows a `useState` hook being set with an initial value from an
+`initialText` prop.
+
+```jsx
+const FancyInput: React.FC<Props> = ({ initialText }) => {
```

Comment:
I would probably start off with an example that doesn’t show it coming from a prop, because otherwise it feels like your state always needs to come from a prop.

Reply:
done